pykern/pkcli/ci.py

```diff
@@ -8,59 +8,52 @@
 :copyright: Copyright (c) 2022 RadiaSoft LLC.  All Rights Reserved.
 :license: http://www.apache.org/licenses/LICENSE-2.0.html
 """
-from pykern.pkdebug import pkdp, pkdlog
-from pykern import pkunit
-from pykern.pkcli import test
+from pykern import pkcli
+from pykern import pkio
 from pykern import pksetup
+from pykern import pkunit
+from pykern.pkcollections import PKDict
+from pykern.pkdebug import pkdp, pkdlog
 import re
 
-
-_FILE_TYPE = re.compile(r".py$")
-_EXCLUDE_FILES = re.compile(
-    f".*(?:{pkunit.DATA_DIR_SUFFIX}|{pkunit.WORK_DIR_SUFFIX})/"
-    + f"|^\\w+/{pksetup.PACKAGE_DATA}/"
-    + r"|pkdebug[^/]*\.py$"
-    + r"|(?:^|/)\."
-    + r"|^run/"
-)
 _PRINT = re.compile(r"(?:\s|^)(?:pkdp|print)\(")
 _PRINT_OK = re.compile(r"^\s*#\s*(?:pkdp|print)\(")
 
 
+def check_eof_newline():
+    """Recursively check repo for files missing newline at end of file.
+
+    Files matching _CHECK_EOF_NEWLINE_FILE_EXTS and not matching
```

Comment:
comment refers to non-existent vars. Better to describe what it does in plain English.

Reply:
7ad30ac